nodejs/eks/nodegroup.ts

```diff
@@ -391,59 +426,68 @@ ${customUserData}
-    if (args.amiId === undefined) {
-        const version = pulumi.output(args.version || core.cluster.version);
-        amiId = version.apply(v => {
-            const filters: { name: string; values: string[] }[] = [
-                // Filter to target Linux arch
-                {
-                    name: "description",
-                    values: ["*linux*", "*Linux*"],
-                },
-                // Filter to target EKS Nodes
-                {
-                    name: "name",
-                    values: ["amazon-eks-node-*"],
-                },
-                // Filter to target General, non-GPU image class
-                {
-                    name: "manifest-location",
-                    values: ["*amazon-eks-node*"],
-                },
-                // Filter to target architecture
-                {
-                    name: "architecture",
-                    values: ["x86_64"],
-                },
-                // Filter to target a specific k8s version
-                {
-                    name: "description",
-                    values: ["*k8s*" + v + "*"],
-                },
-            ];
-
-            // `602401143452` is the ID assigned to `Amazon` and is required for AMIs that are hosted in AWS default regions
-            // `800184023465` is the owner ID of the creator of the EKS Optimized AMI in ap-east-1
-            // `558608220178` is the owner ID of the creator of the EKS Optimized AMI in me-south-1
-            // It is important to note that neither `800184023465` nor `558608220178` are assigned any IDs in default regions
-            // and `602401143452` is not assigned any AMIs in `ap-east-1` or `me-south-1` regions so this lookup is safe to make
-            // AWS Guide for EKS Optimized AMIs https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html
-            const eksWorkerAmiIds = aws.getAmiIds({
-                filters,
-                owners: ["602401143452", "800184023465", "558608220178"],
-                sortAscending: true,
-            }, { parent, async: true });
-
-            const bestAmiId = eksWorkerAmiIds.then(r => r.ids.pop()!);
-            if (!bestAmiId) {
-                throw new Error("No Linux AMI Id was found.");
-            }
-            return bestAmiId;
-        });
-        // When we automatically pick an image to use, we want to ignore any changes to this later by default.
-        ignoreChanges = ["imageId"];
+    if (!args.amiId) {
+        if (args.useLatestAmi || args.useLatestAmiWithGpu) {
+            // https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html
+            amiId = version.apply(v => {
+                const amiType = args.useLatestAmiWithGpu ? "amazon-linux-2-gpu" : "amazon-linux-2";
+                const parameterName = `/aws/service/eks/optimized-ami/${v}/${amiType}/recommended/image_id`;
+                return aws.ssm.getParameter({name: parameterName}).value;
+            });
```

Comment:
Yes, both return the same result today.

> Isn't it just a more reliable way to compute the same thing? What would break?

The thought was related to the filtered search sorting AMIs by the latest image created as the suggested image, and how in the past latest creation date does not always translate to latest recommended by EKS.

But as your comment below highlights, `ignoreChanges` is enabled on this so the thought is moot, and renders this path unnecessary so we'll clean it out.

> The one new thing here is the ability to opt into GPU. Should that just be a gpu: bool property on the NodeGroup?

Yes, good idea. That creates a better UX than the separate opts, and the opts become additive which feels right.